bin/ts-typecheck-all.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+log() {
+    echo "$@" >&2
+}
+
+npm_install() {
+    # --ignore-scripts because we don't want to run any pre- or postinstall scripts
+    # --no-package-lock because we don't want to update or create the package-lock.json
+    # --no-fund because we don't want to check for funding
+    # --no-audit because we don't want to run an audit
+    # --suppress-warnings because we don't want to see any warnings whilst type checking
+    npm i \
+        --suppress-warnings \
+        --ignore-scripts \
+        --no-package-lock \
+        --no-fund \
+        --no-audit \
+        "$@"
+}
+
+main() {
+    local root_dir
+    root_dir=$(git rev-parse --show-toplevel)
+    local dir
+
+    for dir in $(git ls-files samples | grep 'tsconfig.json$' | xargs -n1 dirname | sort -u); do
+        if [[ ! -f "${root_dir}/${dir}/.openapi-generator-ignore" ]]; then
+            # This is not a generated sample; skip it
+            continue
+        fi
+        if [[ ! -f "${root_dir}/${dir}/package.json" ]]; then
+            # we can't really guarantee that all dependencies are there to do a typecheck...
+            continue            
+        fi
+        log "➤ ${dir}"
+        pushd "${root_dir}/${dir}" > /dev/null
+        npm_install \
+            || npm_install --force # --force because we have some incompatible peer-dependencies that can't be fixed
```

Comment:
you can see in the CI logs, but basically there are a couple really really outdated dependencies, which can't even be updated anymore, like https://github.com/angular/tsickle/ for example. Hence we need to sometimes override incompatible peer dependencies, etc.